src/swap/saga.ts

```diff
@@ -467,8 +468,36 @@ export function* swapSubmitPreparedSaga(action: PayloadAction<SwapInfoPrepared>)
 
     const feeCurrencyId = getTokenId(networkId, getFeeCurrency(preparedTransactions))
 
-    const swapTxHash = txHashes[txHashes.length - 1]
+    // if there is an approval transaction, add a standby transaction for it
+    if (preparedTransactions.length > 1) {
+      const approvalTx = preparedTransactions[0]
+      const { functionName, args } = yield* call(decodeFunctionData, {
+        abi: erc20.abi,
+        data: approvalTx.data ?? '0x0',
+      })
+      if (functionName === 'approve' && approvalTx.to === fromToken.address && args) {
+        const approvedAmountInSmallestUnit = args[1] as bigint
+        const approvedAmount = new BigNumber(approvedAmountInSmallestUnit.toString())
+          .shiftedBy(-fromToken.decimals)
+          .toNumber()
```

Comment:
@bakoushin do you remember any specific reason that we went with Decimal and not TokenAmount for `approvedAmount` in the token approval transactions? if there's no reason / preference, i can make a change to use TokenAmount. (although our use case right now doesn't require the tokenId)

Reply:
@kathaypacific yes, it was just the for sake of brevity, because `TokenAmount` basically duplicates all the values we already have. I considered making it consistent, but I wasn't sure that it will worth the complexity. So I went for the simpler solution. If now we see that a consistent solution would be a better fit, let's go for it for sure.

Reply:
Ok the problem is now we use different types: number vs string 
where it should always be string.

Reply:
for now i've fixed the type to be a string (that was my bad, I wrongly interpreted Decimal as a number type) to unblock this PR